### DISCUSSION TITLE
[ZK Stack] - annot find function `exists` in module `fs`

### DISCUSSION THREAD

**Discussion by virgil2019** [zkSync-Community-Hub/zksync-developers#840] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [x] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.79.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

```
zkstackup
```

### Expected Behavior

Compile successfully

### Current Behavior

Compiling zksync_contracts v0.1.0 (/home/ubuntu/.cargo/git/checkouts/zksync-era-657998444dcd9ed9/3cffdb2/core/lib/contracts)
error[E0425]: cannot find function `exists` in module `fs`
   --> /home/ubuntu/.cargo/git/checkouts/zksync-era-657998444dcd9ed9/3cffdb2/core/lib/contracts/src/lib.rs:320:12
    |
320 |     if fs::exists(&bytecode_path)
    |            ^^^^^^ not found in `fs`

error[E0425]: cannot find function `exists` in module `fs`
   --> /home/ubuntu/.cargo/git/checkouts/zksync-era-657998444dcd9ed9/3cffdb2/core/lib/contracts/src/lib.rs:327:16
    |
327 |         if fs::exists(&bytecode_path_legacy)
    |                ^^^^^^ not found in `fs`

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
The error you're encountering indicates that the `exists` function is not found in the `fs` module. This is likely because the Rust standard library does not include an `exists` function directly under `fs`. Instead, you should use the `Path` struct from the `std::path` module to check for the existence of a file.

Here’s how you can modify your code:

1. Import the `Path` struct:
   ```rust
   use std::path::Path;
   ```

2. Replace `fs::exists` with `Path::new` and `exists`:
   ```rust
   if Path::new(&bytecode_path).exists() {
       // Your code here
   }

   if Path::new(&bytecode_path_legacy).exists() {
       // Your code here
   }
   ```

This change should resolve the compilation errors related to the `exists` function. If you encounter further issues or need additional assistance, feel free to ask!

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by kaymomin** [comment]
Hi @virgil2019, thanks for raising the issue. I'll investigate and get back to you asap.

**Reply by kaymomin** [reply]
@virgil2019 Did you try reinstalling/updating all the pre-requisite dependencies? If not, please:

1. Make sure your machine satisfies the system requirements for running the project. You can find the detailed requirements [here](https://github.com/matter-labs/era-compiler-solidity/tree/main#system-requirements).

2. Follow the instructions in the [Setup Dev Guide](https://github.com/matter-labs/zksync-era/blob/3cffdb2d5e144f2e3d8617fa22aacf6cce5998a2/docs/src/guides/setup-dev.md) to install or update the dependencies for the zksync-era repository.

3. Make sure you have the latest version of foundry-zksync installed -> [installation guide](https://foundry-book.zksync.io/getting-started/installation).

Once the above steps are complete, try running the `zkstackup` command again.

Let me know how it goes or if you encounter any further issues.

**Reply by virgil2019** [reply]
I does not work

**Comment by sachanayush47** [comment, marked as answer]
The problem is with rust compiler. Upgrade it to `1.82.0 ` and then try again.

**Reply by kaymomin** [reply]
@virgil2019 checking if this worked for you?

**Reply by virgil2019** [reply]
It works for me
